packages/data/src/redux-store/metadata/test/selectors.js

```diff
@@ -1,117 +1,329 @@
 /**
- * External dependencies
+ * WordPress dependencies
  */
-import EquivalentKeyMap from 'equivalent-key-map';
+import { createRegistry } from '@wordpress/data';
 
-/**
- * Internal dependencies
- */
-import {
-	getIsResolving,
-	hasStartedResolution,
-	hasFinishedResolution,
-	isResolving,
-} from '../selectors';
+jest.useRealTimers();
+jest.setTimeout( 1000000 );
+const testStore = {
+	reducer: ( state = null, action ) => {
+		if ( action.type === 'RECEIVE' ) {
+			return action.items;
+		}
+
+		return state;
+	},
+	selectors: {
+		getFoo: ( state ) => state,
+	},
+};
+
+async function resolve( registry, selector ) {
+	const resolution = registry.resolveSelect( 'store' )[ selector ]();
+	try {
+		await resolution;
+	} catch ( e ) {}
```

Comment:
Nit: styling suggestion.

```suggestion
	try {
		await registry.resolveSelect( 'store' )[ selector ]();
	} catch ( error ) {}
```